File: mobile_app/lib/models/task.dart
/// A task inside a project.
class Task {
  final int id;
  final String title;
  final String status; // todo | in_progress | done
  final int? assignedTo;
  final String? dueDate;

  Task({
    required this.id,
    required this.title,
    required this.status,
    this.assignedTo,
    this.dueDate,
  });

  factory Task.fromJson(Map<String, dynamic> json) => Task(
        id: json['id'] as int,
        title: (json['title'] ?? '') as String,
        status: (json['status'] ?? 'todo') as String,
        assignedTo: json['assigned_to'] as int?,
        dueDate: json['due_date'] as String?,
      );
}

File: app/Models/Task.php
<?php

namespace App\Models;

use App\Models\Concerns\BelongsToTenant;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

/**
 * A task inside a project, owned by a tenant.
 */
class Task extends Model
{
    use HasFactory;
    use BelongsToTenant;

    protected $fillable = [
        'project_id',
        'title',
        'status',       // todo | in_progress | done
        'assigned_to',
        'due_date',
    ];

    protected $casts = [
        'due_date' => 'date',
    ];

    // The project this task belongs to.
    public function project(): BelongsTo
    {
        return $this->belongsTo(Project::class);
    }

    // The tenant user this task is assigned to (optional).
    public function assignee(): BelongsTo
    {
        return $this->belongsTo(TenantUser::class, 'assigned_to');
    }
}
